### PR TITLE
Collective mind

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

While new sprites for the cult of flesh are being drawn, I decided to add other systems that are needed for it. So this is a collective mind that can be used for some creatures like the Dragon, carp, xenos, and as a variant of the dion. The chat can be seen by admins, and it is also impossible to determine who the author of the message is, because it is a "voice in the head."

**Media**

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: VigersRay
- add: Added collective mind for dionas.

### CODE REVIEW

**Comment by ficcialfaint** [comment]
what's the point for admins to see the chat if they won't know who exactly sent it, will they?
Also I think the CL isn't player-facing

**Comment by VigersRay** [comment]
> what's the point for admins to see the chat if they won't know who exactly sent it, will they? Also I think the CL isn't player-facing

Admins see who sent the message...

File: Content.Client/Content.Client.csproj [hunk 1]
```diff
@@ -34,6 +34,9 @@
       <DependentUpon>StrippingWindow.xaml</DependentUpon>
     </Compile>
   </ItemGroup>
+  <ItemGroup>
+    <Folder Include="Flesh\" />
+  </ItemGroup>
```
Comment (ficcialfaint) [inline Content.Client/Content.Client.csproj:39]: Why?

**Comment by ficcialfaint** [comment]
> > what's the point for admins to see the chat if they won't know who exactly sent it, will they? Also I think the CL isn't player-facing
> 
> 
> 
> Admins see who sent the message...

then make a normal description. It said admins can see the chat, but cannot determine the author.
> The chat can be seen by admins, and it is also impossible to determine who the author of the message is

**Comment by metalgearsloth** [comment]
I don't think adding a specific hardcoded channel is the way to go for this kind of stuff as it can quickly spiral out of copy-paste (e.g. lings, xenos, etc) and should be more generic.

**Comment by VigersRay** [comment]
> I don't think adding a specific hardcoded channel is the way to go for this kind of stuff as it can quickly spiral out of copy-paste (e.g. lings, xenos, etc) and should be more generic.

Any suggestions on how best to implement this? Of course, earlier I did an option with just adding radio components to the creature, but in this way its message was heard as a whisper and the name was visible in the chat.

**Review by 0x6273 — CHANGES_REQUESTED** [review]
New .cs files should use file-scoped namespace

**Comment by Froozigiusz** [comment]
What is the correct „official” plural word for diona anyway? Dionas, Dionae or Dionæ? Wiki pages of SS13 servers list the second version most of the time, but I interpret this as an inability to put out æ letter through a default keyboard layout.

File: Content.Client/Chat/CollectiveMindChatUpdateSystem.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<CollectiveMindComponent, ComponentInit>(OnInit);
+            SubscribeLocalEvent<CollectiveMindComponent, ComponentRemove>(OnRemove);
+        }
+
+        public CollectiveMindComponent? Player => CompOrNull<CollectiveMindComponent>(_playerManager.LocalPlayer?.ControlledEntity);
+        public bool IsCollectiveMind => Player != null;
```
Comment (AJCM-git) [inline Content.Client/Chat/CollectiveMindChatUpdateSystem.cs:20]: dont store fields in systems like this, specially if you are gonna reference them in another place, make a method.

File: Content.Server/Chat/Commands/CollectiveMindCommand.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+using Content.Server.Chat.Systems;
+using Content.Shared.Administration;
+using Robust.Server.Player;
+using Robust.Shared.Console;
+using Robust.Shared.Enums;
+
+namespace Content.Server.Chat.Commands
+{
+    [AnyCommand]
+    internal sealed class CollectiveMindCommand : IConsoleCommand
```
Comment (AJCM-git) [inline Content.Server/Chat/Commands/CollectiveMindCommand.cs:10]: try updating this to the new `ToolshedCommand` API, its a lil recent so feel free to ask anything

File: Content.Client/UserInterface/Systems/Chat/ChatUIController.cs [hunk 4]
```diff
@@ -505,6 +507,14 @@ private void UpdateChannelPermissions()
             FilterableChannels |= ChatChannel.AdminAlert;
             FilterableChannels |= ChatChannel.AdminChat;
             CanSendChannels |= ChatSelectChannel.Admin;
+            FilterableChannels |= ChatChannel.CollectiveMind;
```
Comment (AJCM-git) [inline Content.Client/UserInterface/Systems/Chat/ChatUIController.cs:510]: move this one line up to not break the style

File: Content.Server/Chat/Systems/ChatSystem.cs [hunk 5]
```diff
@@ -295,6 +300,56 @@ private void OnGameChange(GameRunLevelChangedEvent ev)
...
+        _chatManager.ChatMessageToManyFiltered(clients,
+            ChatChannel.CollectiveMind,
+            message,
+            messageWrap,
+            source,
+            hideChat,
+            true,
+            sourseCollectiveMindComp.ChannelColor);
+
+        _chatManager.ChatMessageToMany(ChatChannel.CollectiveMind,
```
Comment (AJCM-git) [inline Content.Server/Chat/Systems/ChatSystem.cs:343]: just add `admins` to `clients` and use a single method (you can change the messageWrap with a ternary operator)

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Make your namespaces file scoped

File: Content.Client/UserInterface/Systems/Chat/ChatUIController.cs (hunk 4 above)
Comment (metalgearsloth) [reply Content.Client/UserInterface/Systems/Chat/ChatUIController.cs:510]: We really shouldn't be hardcoding in new channels like this for particular gamemodes tbh, it's just going to become an even bigger mess.

**Comment by AJCM-git** [comment]
@VigersRay are you coming back to this?

**Pull request closed without merging**